### ISSUE TITLE
Support for `aarch64-pc-windows-gnullvm`

### ISSUE THREAD

**Issue by mcandre** [rust-lang/rustup#3297]
### Problem you are trying to solve

Can we get target triples going to support Cygwin-style Rust applications for ARM Windows?

While we do technically have `aarch64-pc-windows-msvc` available, cross-compiling Windows Rust apps from non-Windows hosts is remarkably easier with the `windows-gnu` target triple variants. So a `aarch-pc-windows-gnu` target may be easier for many macOS/Linux/UNIX Rust devs to support than `aarch64-pc-windows-msvc`.

### Solution you'd like

Introduce `aarch-pc-windows-gnu` to the set of available rustup targets, similar to the other `windows-gnu` targets.

**Comment by ChrisDenton** [comment]
I think that this would be an issue for the main rust repository. Rustup just distributes whatever prebuilt targets exist.

And if you look on the [platform support page](https://doc.rust-lang.org/nightly/rustc/platform-support.html/) you'll see there's no `aarch64-pc-windows-gnu` target.

There is a new `aarch64-pc-windows-gnullvm` target (gnu with llvm linker and other tools) but that's currently tier 3 so the rust repository doesn't distribute builds. It would need to be promoted to tier 2 before that can happen.

**Comment by rami3l** [comment]
Closed by https://github.com/rust-lang/rustup/pull/3840.

**Comment by rami3l** [comment]
Sorry, that was a misclick: https://doc.rust-lang.org/nightly/rustc/platform-support/pc-windows-gnullvm.html says it's now tier 2, but without host tools. We'll keep an eye on it.

**Comment by djc** [comment]
IMO since this is not actionable from our side, closing it makes more sense to me -- it can be reopened (or we can work in a new issue) once host tools are available for the target.

**Comment by rami3l** [comment]
> IMO since this is not actionable from our side, closing it makes more sense to me -- it can be reopened (or we can work in a new issue) once host tools are available for the target.

Makes sense; will close it for now until `rustc` and friends say otherwise.

**Comment by benjyw** [comment]
FWIW `aarch64-pc-windows-gnullvm` is now tier 2 with host tools, per https://doc.rust-lang.org/nightly/rustc/platform-support.html

**Comment by ChrisDenton** [comment]
Is there anything for us to do here?

**Comment by rami3l** [comment]
@ChrisDenton I see at least two: adjust the CI to include that target; adjust the shell script to allow detection for it.

**Comment by rami3l** [comment]
Hmmm I think this would be a good opportunity to boost community engagement. I think with a bit of inspiration from https://github.com/rust-lang/rustup/pull/4290 this can be done rather easily even for an external contributor. I'll leave it as an `E-easy`?
